File: 2387-partition-array-such-that-maximum-difference-is-k/2387-partition-array-such-that-maximum-difference-is-k.cs
public class Solution {
     public int PartitionArray(int[] nums, int k)
 {
     Array.Sort(nums);
     int n = nums.Length, last = nums[n - 1], sub = 0, ans = 1;
     for (int i = n - 2; i >= 0; i--)
     {
         if (last - nums[i] + sub > k)
         {
             sub = 0;
             ans++;
         }else
         {
             sub += last - nums[i];
         }
         last = nums[i];
     }
     return ans;
 }
}

File: 2387-partition-array-such-that-maximum-difference-is-k/2387-partition-array-such-that-maximum-difference-is-k.cpp
class Solution {
public:
    int partitionArray(vector<int>& nums, int k) {
	
    sort(nums.begin(), nums.end());
    int n = nums.size(), ans = 1, last = nums[n-1], div = 0;
    for (int i = n - 2; i >= 0; i--)
    {
        if (last - nums[i] + div > k)
        {
            div = 0;
            ans++;
        }
        else
        {
            div += last - nums[i];
        }
        last = nums[i];
    }
    return ans;

}
};